compiler/next/lib/parsing/chpl.ypp

```diff
@@ -1113,76 +1113,89 @@ use_stmt:
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::EXCEPT,
                                      $7);
   }
 | access_control TUSE expr TAS TUNDERSCORE TEXCEPT except_ls TSEMI
   {
     auto ident = toOwned(context->buildIdent(@5, $5));
     auto as = context->buildAsExpr(@3, @5, toOwned($3), std::move(ident));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::EXCEPT,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::EXCEPT,
                                      $7);
   }
 | access_control TUSE expr TONLY opt_only_ls TSEMI
   {
-    auto locUseClause = context->makeSpannedLocation(@3, @5);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned($3),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @5);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned($3),
+                                     VisibilityClause::ONLY,
                                      $5);
   }
 | access_control TUSE expr TAS expr TONLY opt_only_ls TSEMI
   {
     auto as = context->buildAsExpr(@3, @5, toOwned($3), toOwned($5));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::ONLY,
                                      $7);
   }
 | access_control TUSE expr TAS TUNDERSCORE TONLY opt_only_ls TSEMI
   {
     auto ident = toOwned(context->buildIdent(@5, $5));
     auto as = context->buildAsExpr(@3, @5, toOwned($3), std::move(ident));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::ONLY,
                                      $7);
   }
 ;
 
 import_stmt:
   access_control TIMPORT import_ls TSEMI
-    {
-      $$ = TODOSTMT(@$);
-      context->visibility = Decl::DEFAULT_VISIBILITY;
-    }
+  {
+    auto comments = context->gatherComments(@1);
+    auto node = Import::build(BUILDER, LOC(@$), $1, context->consumeList($3));
```

Comment:
Seems like you should be passing the visibility into the build call? The visibility is stored in context->visibility. We have a ParserContext function for building the Use, maybe add one for Import too?

Reply:
I'm confused, I thought in this case the visibility is specified by the `access_control` tag.

Reply:
Yes, you are right about that, my bad. Nevermind!